### PR TITLE
fix: require node-gyp@^8.4.0 & handle old Electron versions

### PR DESCRIPTION
**Pull request by zcbenz**
Since node-gyp >= 8.4.0, building modules for old versions of Electron requires passing `--force-process-config` due to them lacking a valid config.gypi in their headers.

See also https://github.com/nodejs/node-gyp/pull/2497.

### CODE REVIEW

**Review by malept — COMMENTED** [review]
Can we start adding tests for `src/module-type/node-gyp.ts`? You'd be adding `test/module-type-node-gyp.ts` and testing when `--force-process-config` is added to the args.

**Review by malept — CHANGES_REQUESTED** [review]
Going to actually block on the testing request.

**Comment by zcbenz** [comment]
I have updated the code to add tests.

File: src/module-type/node-gyp.ts [hunk 1]
```diff
@@ -36,6 +36,17 @@ export class NodeGyp extends NativeModule {
...
     }
 
+    // Headers of old Electron versions do not have a valid config.gypi file
+    // and --force-process-config must be passed to node-gyp >= 8.4.0 to
+    // correctly build modules for them.
+    // See also https://github.com/nodejs/node-gyp/pull/2497
+    const [major, minor] = this.rebuilder.electronVersion.split('.').slice(0, 2).map(n => parseInt(n, 10));
+    if ((major <= 13) ||
+        (major == 14 && minor <= 1) ||
+        (major == 15 && minor <= 2)) {
```
Comment (malept) [inline src/module-type/node-gyp.ts:46]: @zcbenz what do you think of using `semver` here instead, so that this is more readable? It's already a transitive dependency due to `node-abi`, among other dependencies. Proposed code (assuming I have the logic correct):
```suggestion
    if (!semver.satisfies(this.rebuilder.electronVersion, '^14.2.0 || ^15.3.0 || >= 16.0.0-alpha.1')) {

```

I made the change locally and tests still pass.

**Review by malept — COMMENTED** [review]
I've refactored the tests so that they should test all of the corner cases (and sped them up). But I have an implementation question which could potentially make this a bit easier to maintain.

File: src/module-type/node-gyp.ts (hunk 1 above)
Comment (zcbenz) [reply src/module-type/node-gyp.ts:46]: Using `semver` makes the code looks much better!

**Review by malept — APPROVED** [review]
